epoch1/cuda/ee_mumu/SubProcesses/P1_Sigma_sm_epem_mupmum/check.cc

```diff
@@ -421,30 +436,51 @@ int main(int argc, char **argv)
 
     for (int ievt = 0; ievt < nevt; ++ievt) // Loop over all events in this iteration
     {
-      if (verbose)
-      {
-        // Display momenta
-        const int ipagM = ievt/neppM; // #eventpage in this iteration
-        const int ieppM = ievt%neppM; // #event in the current eventpage in this iteration
-        std::cout << "Momenta:" << std::endl;
+      auto getMomentum = [&](std::size_t evtNo, int particle, int component) {
+        assert(component < mgOnGpu::np4);
+        assert(particle  < mgOnGpu::npar);
+        const auto page  = evtNo / neppM; // #eventpage in this iteration
+        const auto ieppM = evtNo % neppM; // #event in the current eventpage in this iteration
+        return hstMomenta[page * npar*np4*neppM + particle * neppM*np4 + component * neppM + ieppM];
+      };
+      auto dumpParticles = [&](std::ostream& stream, std::size_t evtNo, unsigned precision) {
+        const auto width = precision + 8;
```

Comment:
I trust you on this one :-)
I am not sure if there is a need to generalise, but maybe you need it in the file dump